python/cudf/cudf/core/indexed_frame.py

```diff
@@ -55,6 +56,13 @@
 from cudf.utils import docutils
 from cudf.utils.utils import _cudf_nvtx_annotate
 
+if sys.version_info >= (3, 10):
+    # Introduced in Python 3.10
+    from types import NotImplementedType
+else:
+    NotImplementedType = Any
```

Comment:
Rather than repeating this logic everywhere, let's move it to [`cudf._typing`](https://github.com/rapidsai/cudf/blob/branch-22.10/python/cudf/cudf/_typing.py).

Reply:
Done.